traitlets/config/application.py

```diff
@@ -381,7 +404,7 @@ def print_help(self, classes=False):
         self.print_options()
 
         if classes:
-            help_classes = self.classes
+            help_classes = self._classes_in_config_sample()
```

Comment:
See #289 discussion, and my example just entered in #180, specifically the comment in `classes` trait.